### PR TITLE
Feature/1238/266 Verify OpenAPI definition files

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

-  Make OpenAPI output reproducible (sorted)
-  Update CI to fail if OpenAPI generated output doesn't match committed files. Outputs diff in build logs in case of errors

## Why it does that

Repository contains OpenAPI YAML definitions generated from files. These must be manually updated and can go out of sync.

## Further notes

Moved check job to same workflow as all other checks (`verify.yaml`) for consistency.

Need to post-process YAML to sort it to obtain a comparable diff. Created PR in plugin to solve this: https://github.com/kpramesh2212/openapi-merger-plugin/pull/11

Sample failing job: https://github.com/agera-edc/DataSpaceConnector/runs/6443856078?check_suite_focus=true

## Linked Issue(s)

#266 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: .github/workflows/verify.yaml [hunk 1]
```diff
@@ -49,6 +49,31 @@ jobs:
...
+  OpenAPI-Definitions:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '17'
+
+      - name: Save previous OpenAPI definition
+        run: yq -P 'sort_keys(..)' resources/openapi/openapi.yaml > openapi-git-sorted.yaml
```
Comment (chrislomonico) [inline .github/workflows/verify.yaml:61]: Is yq installed by default on the ubuntu-latest distro?

**Review by chrislomonico — COMMENTED** [review]
just a general question regarding `yq` on the `ubuntu-latest` image

File: .github/workflows/verify.yaml (hunk 1 above)
Comment (algattik) [reply .github/workflows/verify.yaml:61]: Yes, as demonstrated by the build respectively succeeding, or failing with diff

File: .github/workflows/verify.yaml [hunk 2]
```diff
@@ -49,6 +49,31 @@ jobs:
...
+          ./gradlew resolve
+          ./gradlew mergeOpenApiFiles
+
+      # Can be used (and yq tasks removed) when https://github.com/kpramesh2212/openapi-merger-plugin/pull/11/files is merged
+      #- name: Check OpenAPI definitions match code
+      #  run: git diff --exit-code
+
+      - name: Check OpenAPI definitions match code
+        run: |
+          yq -P 'sort_keys(..)' resources/openapi/openapi.yaml > openapi-sorted.yaml
```
Comment (cpeeyush) [inline .github/workflows/verify.yaml:74]: nit: just the name of sorted file. Ignore if not agree. Thanks.
```suggestion
          yq -P 'sort_keys(..)' resources/openapi/openapi.yaml > openapi-ci-sorted.yaml
```

File: .github/workflows/verify.yaml [hunk 3]
```diff
@@ -49,6 +49,31 @@ jobs:
...
+          diff openapi-git-sorted.yaml openapi-sorted.yaml
```
Comment (cpeeyush) [inline .github/workflows/verify.yaml:75]: adjusting accordingly, if we agree to take above suggestion.
```suggestion
          diff openapi-git-sorted.yaml openapi-ci-sorted.yaml
```

**Review by cpeeyush — APPROVED** [review]

**Review by chrislomonico — APPROVED** [review]

**Pull request merged**